### PR TITLE
Feature/homepage load more

### PR DESCRIPTION
**Pull request by jacoblipech**

### CODE REVIEW

File: src/main/webapp/assets/js/index.js [hunk 1]
```diff
@@ -98,20 +101,39 @@ function filterCategory(category) {
...
+    } else {
+      productsToLoad[i].style.display = "none";
+    }
+  }
+
+  // show load more button
+  if (numProductsLoaded < productsToLoad.length) {
+    document.getElementById("loadMore").style.display = "";
+  }
+  products = productsToLoad
```
Comment (g-lilian) [inline src/main/webapp/assets/js/index.js]: Think you missed a ; here

File: src/main/webapp/assets/js/index.js [hunk 2]
```diff
@@ -98,20 +101,39 @@ function filterCategory(category) {
...
     let existingCategories = [];
-    categoryPills = products[i].getElementsByClassName('category-pill');
+    categoryPills = fullProducts[i].getElementsByClassName('category-pill');
     for (let i = 0; i < categoryPills.length; i++) {
       existingCategories.push(categoryPills[i].innerHTML);
     }
+    // only show listings within the category
     if (existingCategories.includes(categorySelected) || categorySelected == "All") {
-      products[i].style.display = "";
+      productsToLoad.push(fullProducts[i])
```
Comment (g-lilian) [inline src/main/webapp/assets/js/index.js]: Missing ;

File: src/main/webapp/assets/js/index.js [hunk 3]
```diff
@@ -98,20 +101,39 @@ function filterCategory(category) {
   document.getElementById(`filter-btn-${category}`).className += " active";
 
   // filter for products
-  let products = document.getElementsByClassName("product-listing-card");
+  let fullProducts = document.getElementsByClassName("product-listing-card");
+  var productsToLoad = []
```
Comment (g-lilian) [inline src/main/webapp/assets/js/index.js]: Missing ; also. I'm kind of surprised js doesn't throw compile errors for this?

**Review by g-lilian — APPROVED** [review]
Generally lgtm!

File: src/main/webapp/assets/js/index.js (hunk 3 above)
Comment (jirehcwe) [reply src/main/webapp/assets/js/index.js]: Javascript has "optional" semicolons. it's weird.

**Pull request merged**